climpred/classes.py

```diff
@@ -40,34 +40,38 @@ def _display_metadata(self):
     """
     header = f'<climpred.{type(self).__name__}>'
     summary = header + '\nInitialized Ensemble:\n'
-    summary += '    ' + str(self.initialized.data_vars)[18:].strip() + '\n'
+    summary += '    ' + str(self._datasets['initialized'].data_vars)[18:].strip() + '\n'
     if isinstance(self, HindcastEnsemble):
-        if any(self.reference):
-            for key in self.reference:
+        if any(self._datasets['reference']):
+            for key in self._datasets['reference']:
                 summary += f'{key}:\n'
-                N = len(self.reference[key].data_vars)
+                N = len(self._datasets['reference'][key].data_vars)
                 for i in range(1, N + 1):
                     summary += (
                         '    '
-                        + str(self.reference[key].data_vars).split('\n')[i].strip()
+                        + str(self._datasets['reference'][key].data_vars)
+                        .split('\n')[i]
+                        .strip()
                         + '\n'
                     )
         else:
             summary += 'References:\n'
             summary += '    None\n'
     elif isinstance(self, PerfectModelEnsemble):
         summary += 'Control:\n'
-        if any(self.control):
-            N = len(self.control.data_vars)
+        if any(self._datasets['control']):
+            N = len(self._datasets['control'].data_vars)
             for i in range(1, N + 1):
                 summary += (
-                    '    ' + str(self.control.data_vars).split('\n')[i].strip() + '\n'
+                    '    '
+                    + str(self._datasets['control'].data_vars).split('\n')[i].strip()
```

Comment:
I dislike how blacken makes code weirdly formatted like this and also inconsistent with above: .split().strip() was placed on a new line

Reply:
It's not perfect, but it at least automates the formatting so it's not a debate that needs to be had with contributors, and it's something that we don't have to think about. It might be splitting differently based on the line length limits. There's a lot of flags on the `black` page if you have any suggestions. But for now I'd rather have the consistency of this than a ton of different quirks from different contributors down the line.